drivers/mbox/mbox_nrf_bellboard_local.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_bellboard_local
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+
+#include <haly/nrfy_bellboard.h>
+
+#define BELLBOARD_IRQ_COUNT 4U
+
+/* BELLBOARD runtime resources for each group */
+struct mbox_bellboard_local_group {
+	mbox_callback_t cb[NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT];
+	void *user_data[NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT];
+	uint32_t enabled_mask;
+};
+
+#define BELLBOARD_GRP_MAYBE_DEFINE(grp)                                                            \
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(0, grp),                                                  \
+		    (static struct mbox_bellboard_local_group g_##grp;), ())
+
+#define BELLBOARD_GRP_GET_OR_NULL(grp) COND_CODE_1(DT_INST_IRQ_HAS_NAME(0, grp), (&g_##grp), (NULL))
+
+BELLBOARD_GRP_MAYBE_DEFINE(bellboard0)
+BELLBOARD_GRP_MAYBE_DEFINE(bellboard1)
```

Comment:
This is a bit confusing as `bellboardX` looks like an instance of the BELLBOARD peripheral. I'd suggest to use `irqX` instead.